packages/mobx/__tests__/v5/base/observables.js

```diff
@@ -2360,3 +2360,30 @@ describe("`requiresObservable` takes precedence over global `reactionRequiresObs
         expect(consoleWarnSpy).not.toHaveBeenCalled()
     })
 })
+
+test("state version updates correctly", () => {
+    // This test was designed around the idea of updating version only at the end of batch,
+    // which is NOT an implementation we've settled on, but the test is still valid.
```

Comment:
Sounds correct to me though :)

Reply:
I am not sure what you mean. Just to provide some background. I've been looking for a common place where I could update state version. `endBatch` (when nesting level gets back to zero) seemed like a good candidate, because it's actually called even for individual mutations, but I was worried about some edge cases, like autorun invoking itself. So I wrote this test and gave it a go. However I quickly realized `endBatch` is also often called without any actual mutation, therefore I threw the idea away and moved the version update to `reportChanged`, but I kept the test as is. I've left the comment to clarify why the test looks this way.